lib/chibios-syscalls/malloc_lock.c

```diff
@@ -1,20 +1,35 @@
 #include <ch.h>
+#include <hal.h>
+
 #include <malloc.h>
 
-static MUTEX_DECL(lock);
```

Comment:
Why remove the static mutex initializer? (MUTEX_DECL() initializes the lock already)
This adds complexity because it requires calling `malloc_lock_init()` instead of letting the data section initializer do the job.
It also adds the overhead of `if (lock_initialized)` check.

Reply:
Agree, MUTEX_DECL is the way to go. I guess its because this implementation was done differently in two boards and salah copied it over ?

Reply:
Just to be clear, this relies on the data section being initialized before any code that can call malloc (such as static constructors). But this is the case anyway, so no problem.

> I guess its because this implementation was done differently in two boards

Yes, it was done this way in master-firmware.

Reply:
> I guess its because this implementation was done differently in two boards and salah copied it over ?

:see_no_evil: yes, of course I had to trust the master-firmware implementation.
I need to test on hardware again :crying_cat_face: